packages/grafana-data/src/transformations/transformers/seriesToColumns.ts

```diff
@@ -68,7 +68,7 @@ export const seriesToColumnsTransformer: DataTransformerInfo<SeriesToColumnsOpti
     });
 
     for (const item of allFields) {
-      resultFrame.addField(item.newField);
+      item.newField = resultFrame.addField(item.newField);
```

Comment:
Maybe we should add the same test case from my PR?

Reply:
yea, I need to add a test case for seriesToColumn, as without the above line things break quite badly

Reply:
added test